packages/camera/camera_android_camerax/android/src/test/java/io/flutter/plugins/camerax/PreviewTest.java

```diff
@@ -0,0 +1,177 @@
+
+    previewHostApi.cameraXProxy = mockCameraXProxy;
+    testInstanceManager.addDartCreatedInstance(mockPreview, 5L);
+
+    when(mockTextureRegistry.createSurfaceTexture()).thenReturn(mockSurfaceTextureEntry);
+    when(mockSurfaceTextureEntry.surfaceTexture()).thenReturn(mockSurfaceTexture);
+    when(mockSurfaceTextureEntry.id()).thenReturn(120L);
+    when(mockSurfaceRequest.getResolution()).thenReturn(new Size(200, 500));
+    when(mockCameraXProxy.createSurface(mockSurfaceTexture)).thenReturn(mockSurface);
+    when(mockCameraXProxy.createSystemServicesFlutterApi(mockBinaryMessenger))
+        .thenReturn(mockSystemServicesFlutterApi);
+
+    final ArgumentCaptor<Preview.SurfaceProvider> surfaceProviderCaptor =
+        ArgumentCaptor.forClass(Preview.SurfaceProvider.class);
+    final ArgumentCaptor<Surface> surfaceCaptor = ArgumentCaptor.forClass(Surface.class);
+    final ArgumentCaptor<Consumer> consumerCaptor = ArgumentCaptor.forClass(Consumer.class);
+
+    // Test that surface provider was set and the surface texture ID was returned.
+    assertEquals((long) previewHostApi.setSurfaceProvider(5L), 120L);
+    verify(mockPreview).setSurfaceProvider(surfaceProviderCaptor.capture());
+
+    Preview.SurfaceProvider surfaceProvider = surfaceProviderCaptor.getValue();
+    surfaceProvider.onSurfaceRequested(mockSurfaceRequest);
+
+    verify(mockSurfaceTexture).setDefaultBufferSize(200, 500);
+    verify(mockSurfaceRequest)
+        .provideSurface(surfaceCaptor.capture(), any(Executor.class), consumerCaptor.capture());
+
+    // Test that the surface derived from the surface texture entry will be provided to the surface request.
+    assertEquals(surfaceCaptor.getValue(), mockSurface);
+
+    // Test that the Consumer used to handle surface request result releases Flutter surface texture appropriately
+    // and sends camera errors appropriately.
+    Consumer<SurfaceRequest.Result> capturedConsumer = consumerCaptor.getValue();
+    when(mockSurfaceRequestResult.getResultCode())
+        .thenReturn(SurfaceRequest.Result.RESULT_SURFACE_USED_SUCCESSFULLY);
+    capturedConsumer.accept(mockSurfaceRequestResult);
+    when(mockSurfaceRequestResult.getResultCode())
+        .thenReturn(SurfaceRequest.Result.RESULT_REQUEST_CANCELLED);
+    capturedConsumer.accept(mockSurfaceRequestResult);
+    when(mockSurfaceRequestResult.getResultCode())
+        .thenReturn(SurfaceRequest.Result.RESULT_INVALID_SURFACE);
+    capturedConsumer.accept(mockSurfaceRequestResult);
+    when(mockSurfaceRequestResult.getResultCode())
+        .thenReturn(SurfaceRequest.Result.RESULT_WILL_NOT_PROVIDE_SURFACE);
+    capturedConsumer.accept(mockSurfaceRequestResult);
+    when(mockSurfaceRequestResult.getResultCode())
+        .thenReturn(SurfaceRequest.Result.RESULT_SURFACE_ALREADY_PROVIDED);
+    capturedConsumer.accept(mockSurfaceRequestResult);
+
+    verify(mockSurface, times(4)).release();
```

Comment:
It might make sense to verify the behavior of each of these in their own tests or at minimum immediately after they get called instead of verifying the combined behavior at the end.  
```
when(mockSurfaceRequestResult.getResultCode())
        .thenReturn(SurfaceRequest.Result.RESULT_SURFACE_USED_SUCCESSFULLY);
    capturedConsumer.accept(mockSurfaceRequestResult);
verify(mockSurface, times(1)).release();
reset(mockSurface);
```

Reply:
I pushed this to a separate test. It's still kinda long but I think it makes more sense because this is specifically testing the separate helper I made as per another comment you left!

Reply:
Consider clearing mock interactions after this line. Below you have `verify(mockSurface, times(5)).release();` when that line below does not actually want to re-verify that release was called in the above cases only that it was called again in RESULT_INVALID_SURFACE.